tests/unit/amo/components/TestSearchForm.js

```diff
@@ -12,52 +12,73 @@ import {
   ADDON_TYPE_EXTENSION,
   ADDON_TYPE_LANG,
   ADDON_TYPE_THEME,
+  CLIENT_APP_ANDROID,
+  CLIENT_APP_FIREFOX,
+  VIEW_CONTEXT_EXPLORE,
   VIEW_CONTEXT_HOME,
 } from 'core/constants';
 import LoadingText from 'ui/components/LoadingText';
 import {
+  autocompleteCancel,
+  autocompleteStart,
+} from 'core/reducers/autocomplete';
+import {
   createFakeAutocompleteResult,
   dispatchAutocompleteResults,
-  dispatchClientMetadata,
   dispatchSignInActions,
 } from 'tests/unit/amo/helpers';
 import {
   createFakeEvent,
   createStubErrorHandler,
   getFakeI18nInst,
 } from 'tests/unit/helpers';
-import {
-  autocompleteCancel,
-  autocompleteStart,
-} from 'core/reducers/autocomplete';
 
 
 describe(__filename, () => {
-  const pathname = '/search/';
-  const api = { clientApp: 'firefox', lang: 'de' };
   let errorHandler;
-  let router;
+  // let fakeDispatch;
+  let fakeRouter;
 
-  const mountComponent = ({
-    store = dispatchClientMetadata().store,
+  function mountComponent({
+    pathname = '/search/',
+    query = 'foo',
+    store = dispatchSignInActions().store,
     ...props
-  } = {}) => {
+  } = {}) {
     return mount(
```

Comment:
why do so many tests need to use `mount()`? Is it just that they haven't been converted to use `shallow()` yet?

Reply:
Isn't it because we use `simulate()` on components that are not `root`? Would `shallowUntilTarget()` allow that?

Reply:
It's because they are checking state, actually. I'll add some comments. A few are using it because they were supplying props directly and not dispatching and it was a fair bit of wrangling to convert them. I can see about migrating the last few.